spec/features/filling_out_application_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+describe 'filling out an application' do
+  let(:application) do
+    create :application_template, :with_questions
+  end
+
+  before :each do
+    when_current_user_is :student, integration: true
+    visit application_path(application)
+  end
+
+  context 'answered all required questions' do
+    it 'redirects to the student dashboard' do
+      click_on 'Submit application'
+      expect(page.current_url).to eql student_dashboard_url
+    end
+
+    it 'renders a flash message' do
+      expect_flash_message(:application_receipt)
+      click_on('Submit application')
```

Comment:
Not something I think needs fixing, but it's worth noting that you can leave both sets of parentheses out of this test:

```ruby
expect_flash_message :application_receipt
click_on 'Submit application'
```

It's a matter of personal preference - mine is to write as few non-alphanumeric characters as possible in my Ruby code.